### DISCUSSION TITLE
BasicTerm_S - disc_factors() & disc_rate_mth()

### DISCUSSION THREAD

**Discussion by ghost** [lifelib-dev/lifelib#48] [General]
Hi,

I'm new to this and have not done modelling for quite a while so I was learning the program by creating a spreadsheet to replicate the numbers. For the first model point, the two functions below have the following output:
disc_factors() : array([1.        , 1.        , 1.        , 1.        , 1.        ,       1.        , 1.        , 1.        , 1.        , 1.        ,       1.        , 1.        , 0.99448063, 0.99402206, 0.9935637 ,...])
disc_rate_mth() : array([0.        , 0.        , 0.        , 0.        , 0.        ,       0.        , 0.        , 0.        , 0.        , 0.        ,       0.        , 0.        , 0.00046133, 0.00046133, 0.00046133,...])
I was wondering if the first eleven 0's from disc_rate_mth() should not be there and disc_factors() be calculated as array([1, (1+i_mth)^-1, (1+i_mth)^-2,...])?

**Comment by fumitoh** [comment]
0 is set from input to indicate the first year (year=0) discount rate is 0. See the discount_curve tab in basic_term_sample.xlsx included in `basiclife`.

**Reply by ghost** [reply]
Thanks for the explanation and pointing me to the spreadsheet. I have another question after looking at the spreadsheet. The following two functions are both using disc_factors() to calculate the present value:

pv_claims() = sum(list(claims(t) for t in range(proj_len())) * disc_factors()[:proj_len()])
pv_pols_if() = sum(list(pols_if(t) for t in range(proj_len())) * disc_factors()[:proj_len()])

For claims, I am assuming end of month payment and for pv_pols_if(), I am assuming beginning of month for premium payment. From the spreadsheet, it seems both are done assuming beginning of month. To explain my question more clearly, if I change year 0 discount rate to be a rate greater than 0, the first element of the disc_factors array remains at 1 due to it being at the time of valuation and the next 11 element being less than 1 due to discounting. The first month claims of 34.18 is multiplied by 1, implying beginning of the month. I was wondering if the first month claims should be multiplied by the second element of the disc_factors array instead and so on?

**Reply by fumitoh** [reply]
Yes, if you want to be precise, but the impact is small, so ignoring one month lag is a simplification.

**Reply by ghost** [reply]
Yes, agreed.
